src/converters/table-post-fixer.js

```diff
@@ -0,0 +1,258 @@
+ * * None of a table cells extends vertically beyond its section (either header or body).
+ *
+ * If the table structure is not correct, the post-fixer will automatically correct it in two steps:
+ *
+ * 1. It will clip table cells that extends beyond it section.
+ * 2. It will add empty table cells to those rows which are narrower then the widest table row.
+ *
+ * ## Clipping overlapping table cells
+ *
+ * Such situation may occur when pasting a table (or part of a table) to the editor from external sources.
+ *
+ * As an example see below table which has a table cell (FOO) with a `rowspan` attribute of `2`:
+ *
+ *		<table headingRows="1">
+ *			<tableRow>
+ *				<tableCell rowspan="2">FOO</tableCell>
+ *				<tableCell colspan="2">BAR</tableCell>
+ *			</tableRow>
+ *			<tableRow>
+ *				<tableCell>BAZ</tableCell>
+ *				<tableCell>XYZ</tableCell>
+ *			</tableRow>
+ *		</table>
+ *
+ * will be rendered in the view as:
+ *
+ *		<table>
+ *			<thead>
+ *				<tr>
+ *					<td rowspan="2">FOO</td>
+ *					<td colspan="2">BAR</td>
+ *				</tr>
+ *			</thead>
+ *			<tbody>
+ *				<tr>
+ *					<td>BAZ<td>
+ *					<td>XYZ<td>
+ *				</tr>
+ *			</tbody>
+ *		</table>
+ *
+ * In above example the table will be rendered as a table with two rows - one in the header and second one in the body.
+ * The table cell with FOO contents will not expand to the body section so its `rowspan` attribute will be changed to `1` (and as
+ * the value `1` is a default value the `rowspan` attribute will be removed from the model).
+ *
+ * The table cell with BAZ contents will be in the first column of a table.
+ *
+ * ## Adding missing table cells
+ *
+ * The table post-fixer will insert empty table cells to equalize table rows lengths. The width of a table row is calculated by counting
+ * widths of table cells and widths of columns spanned by cells from rows above in a given row.
+ *
+ * In the above example the table row in body section of the table is narrower then the row from the header - it has two cells
+ * with default colspan (value of `1`). The header row has one cell with width = 1 and second withe width = 2.
+ * The table cell FOO does not expand beyond head section (and as such will be fixed in the first step of this post-fixer).
+ * The post-fixer will add a missing table cell to the row in body section of the table.
+ *
+ * The table from the above example will be fixed and rendered to the view as below:
+ *
+ *		<table>
+ *			<thead>
+ *				<tr>
+ *					<td rowspan="2">FOO</td>
+ *					<td colspan="2">BAR</td>
+ *				</tr>
+ *			</thead>
+ *			<tbody>
+ *				<tr>
+ *					<td>BAZ<td>
+ *					<td>XYZ<td>
+ *				</tr>
+ *			</tbody>
+ *		</table>
+ *
+ * **Note** The table post-fixer only ensures proper structure without deeper analysis of the nature of a change. As such it might lead
+ * to a structure which was not intended by user changes. In particular it will also fix undo steps (in conjunction with collaboration)
```

Comment:
> by the user

> In particular,

TBH, I'd write a little bit more about the collaboration case. I think it's vital for the documentation.